keras/callbacks.py

```diff
@@ -671,19 +673,25 @@ def set_model(self, model):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        self.saver = tf.train.Saver()
-        self.embeddings = {layer.name: (layer.weights[0],
-                                        os.path.join(self.log_dir, layer.name +
-                                                     '.ckpt'))
-                           for layer in self.model.layers
-                           if layer.name in self.embeddings_layer_names}
-        self.embeddings_metadata_setup()
+        if self.embeddings_freq:
+            self.saver = tf.train.Saver()
+
+            embeddings_layer_names = self.embeddings_layer_names \
+                                     if self.embeddings_layer_names else \
+                                        [layer.name for layer in self.model.layers
+                                         if type(layer).__name__ == 'Embedding']
+            self.embeddings = {layer.name:
+                (layer.weights[0], os.path.join(self.log_dir,
+                                                layer.name + '.ckpt'))
+                               for layer in self.model.layers
+                               if layer.name in embeddings_layer_names}
+            self.embeddings_metadata_setup()
 
     def on_epoch_end(self, epoch, logs=None):
         logs = logs or {}
 
         if self.validation_data and self.histogram_freq:
-            if epoch % self.histogram_freq == 0:
+            if (epoch + 1) % self.histogram_freq == 0:
```

Comment:
As I described in the initial post, `epoch` initial value is 0, not 1, so `epoch` modulo `self.embeddings_freq` for initial epoch is 0. It can be unexpected behaviour. For example, I asked to dump embeddings each 10 epochs, and train model for 20 ones. There will be dumped two epochs, 0-th and 10-th, but I expect 9-th and 19-th to be dumped (i.e. 10-th and 20-th in one-based reckoning).